### PR TITLE
added CustomGlobalExceptionHandler

### PR DESCRIPTION
**Pull request by Serhii-Nosulia**

### CODE REVIEW

File: src/main/java/mate/bookshopapp/exception/CustomGlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            HttpStatusCode status,
+            WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMassage)
+                .toList();
+        body.put("errors", errors);
```
Comment (Sanderfuck) [inline src/main/java/mate/bookshopapp/exception/CustomGlobalExceptionHandler.java]: You can use constants for strings

File: src/main/java/mate/bookshopapp/dto/book/UpdateBookDto.java [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+import jakarta.validation.constraints.Min;
 import java.math.BigDecimal;
 
 public record UpdateBookDto(String title,
                             String author,
                             String isbn,
-                            BigDecimal price,
+                            @Min(0) BigDecimal price,
                             String description,
                             String coverImage) {
```
Comment (Sanderfuck) [inline src/main/java/mate/bookshopapp/dto/book/UpdateBookDto.java]: Only one field validated?

File: src/main/java/mate/bookshopapp/dto/book/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 
-public record CreateBookRequestDto(String title,
-                                   String author,
-                                   String isbn,
-                                   BigDecimal price,
+public record CreateBookRequestDto(@NotNull String title,
+                                   @NotNull String author,
+                                   @NotNull String isbn,
+                                   @NotNull @Min(0) BigDecimal price,
                                    String description,
```
Comment (Sanderfuck) [inline src/main/java/mate/bookshopapp/dto/book/CreateBookRequestDto.java]: Validate description by max size, find validate case for ibn

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: src/main/java/mate/bookshopapp/dto/book/UpdateBookDto.java [hunk 4]
```diff
@@ -1,11 +1,15 @@
...
 import java.math.BigDecimal;
+import mate.bookshopapp.validation.Isbn;
 
-public record UpdateBookDto(String title,
-                            String author,
-                            String isbn,
-                            BigDecimal price,
-                            String description,
+public record UpdateBookDto(@NotNull String title,
+                            @NotNull String author,
```
Comment (okuzan) [inline src/main/java/mate/bookshopapp/dto/book/UpdateBookDto.java]: for strings let's use NotBlank
https://www.baeldung.com/java-bean-validation-not-null-empty-blank

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by okuzan — APPROVED** [review]

**Pull request merged**